csharp/ql/lib/semmle/code/csharp/commons/Collections.qll

```diff
@@ -67,6 +69,42 @@ class CollectionType extends RefType {
   }
 }
 
+/**
+ * A collection type that can be used as a `params` parameter type.
+ */
+abstract private class ParamsCollectionTypeImpl extends ValueOrRefType {
+  /**
+   * Gets the element type of this collection, for example `int` in `IEnumerable<int>`.
+   */
+  abstract Type getElementType();
+}
+
+private class AddArrayType extends ParamsCollectionTypeImpl instanceof ArrayType {
```

Comment:
Why is this class named `AddArrayType`? An alternative would be `ParamsArrayType`.